install_config/configuring_sdn.adoc

```diff
@@ -275,6 +278,17 @@ service neutron-vpn-agent restart
 service neutron-server  restart
 ----
 
+When creating the OpenStack instances, disable both port-security and security
+groups in the ports where the container network flannel interface will be:
+
+----
+neutron port-update $port --no-security-groups --port-security-enabled=False;
+----
+
+NOTE: Flannel requires to gather information from etcd to configure and assign
+the subnets in the nodes, therefore, the security group attached to the master
```

Comment:
do you mean a egress rule to reach etcd nodes, or implying ingress rules for the cases when master is also etcd node?

Reply:
Could you please also note if the flannel network could be isolated from other networks?
For security reasons it would be nice to have it not routed with other networks.

Reply:
So, the nodes needs to query etcd to gather the networking configuration for flannel[1], so you need:
* Egress rules in the nodes security group to allow outbound traffic from nodes to etcd cluster
* Ingress rules in the etcd nodes (masters, or dedicated etcd cluster instances) to allow inbound traffic from nodes to etcd [2](2379/tcp)

We didn't set egress rules in the ref. arch. as egress is open by default

[1] https://github.com/openshift/openshift-ansible/blob/master/roles/flannel/tasks/main.yml#L7:L18
[2] https://access.redhat.com/documentation/en-us/reference_architectures/2017/html-single/deploying_and_managing_red_hat_openshift_container_platform_3.6_on_red_hat_openstack_platform_10/#enable_flannel_etcd_query

Reply:
Regarding security reasons, that's exactly what we do in the ref. arch. by creating a different network.
I can add that, but I'd like to first have a word from the docs. team if they want that too.

Reply:
"Flannel gathers information..." or "Flannel is required to gather information..."